engine/src/main/java/io/zeebe/engine/processing/streamprocessor/ProcessingStateMachine.java

```diff
@@ -217,8 +217,8 @@ private void processEvent(final LoggedEvent event) {
       return;
     }
 
-    metrics.processingLatency(
-        metadata.getRecordType(), event.getTimestamp(), ActorClock.currentTimeMillis());
+    final long processingStartTime = ActorClock.currentTimeMillis();
+    metrics.processingLatency(metadata.getRecordType(), event.getTimestamp(), processingStartTime);
```

Comment:
Can you explain the choice to include deserialization as part of the processing time?

Reply:
As a first step I did not want to narrow the scope of a metrics. If processing duration is higher than what we expect, then we can narrow down the metrics to smaller blocks. Would you like to take deserialization out of the processing time?

Reply:
I mostly wanted to know if it was a conscious choice, and if so, why. Do you think it might be confusing/unexpected that deserializing is part of the processing time?

Reply:
Actually, I wanted to include the complete processing - from this event is ready to process until the next event is ready. This would give us an idea how much time we spent in StreamProcessor. So that should include the steps `updateState` and `writeEvent`, which is not currently included in the processing time. Wdyt? Shall I update it? Then it wouldn't be weird to have deserializing also part of the processing time.

Reply:
That makes more sense - we can make it more granular by adding a metric per step later on (i.e. updateState, writeFollowUpEvents, etc.). If we add one here, can we also add one in reprocessing? Both will be very useful when refactoring how we do stream processing next quarter :+1: